### PR TITLE
fix clipping plane bounding box

### PR DESCRIPTION
**Pull request by basisunus**

### CODE REVIEW

**Comment by dcwhite** [comment]
The "d" slider does not cover the full range of the mesh. Maybe it only works on the 3rd floor of the building...

File: src/Interface/Modules/Render/ES/SRInterface.cc [hunk 1]
```diff
@@ -1275,9 +1275,23 @@ namespace SCIRun {
...
+        //boundbox transformation
+        glm::mat4 trans_bb;
+        glm::vec3 scale_bb(mSceneBBox.x_length() / 2.0, mSceneBBox.y_length() / 2.0, mSceneBBox.z_length() / 2.0);
+        glm::vec3 center_bb(mSceneBBox.center().x(), mSceneBBox.center().y(), mSceneBBox.center().z());
+        trans_bb = glm::scale(trans_bb, scale_bb);
+        trans_bb = glm::translate(trans_bb, center_bb);
         for (auto i : clippingPlanes_)
         {
-          clippingPlanes->clippingPlanes.push_back(glm::vec4(i.x, i.y, i.z, i.d));
+          glm::vec4 o = glm::vec4(i.x, i.y, i.z, 1) * (-i.d);
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ES/SRInterface.cc]: Pseudo code idea for refactoring:
```
for (auto& i : clippingPlanes_)
        {
		i.update();
		clippingPlanes->clippingPlanes[index] = i.normal;
		clippingPlanes->clippingPlanesCtrls[index] = i.ctrlVec;
```
These are new methods on class ClippingPlane.

Then the clippingPlanes/Ctrls vectors can just be assigned to after recomputation of the two vec4s.

**Comment by dcwhite** [comment]
Is this fixed yet?

**Comment by dcwhite** [comment]
:8ball:

**Comment by basisunus** [comment]
It's fixed but in a later branch. It'll be merged along with the text branch.

**Comment by cbrightsci** [comment]
issue #1329 should remain open

**Pull request merged**